tools/mason/MasonSearch.chpl

```diff
@@ -0,0 +1,78 @@
+
+use MasonHelp;
+use MasonUpdate;
+
+use FileSystem;
+use Regexp;
+
+//
+// TODO:
+// - order results by some kind of 'best match' metric
+// - allow regex searches
+// - allow for exclusion of a pattern
+//
+
+proc masonSearch(origArgs : [] string) {
+  var args : [1..origArgs.size] string = origArgs;
+
+  const printHelp = || reduce [a in args] (a == "-h" || a == "--help");
+  if printHelp {
+    masonSearchHelp();
+    exit(0);
+  }
+
+  updateRegistry("", args);
+
+  consumeArgs(args);
+
+  // If no query is provided, list all packages in registry
+  const query = if args.size > 0 then args.tail().toLower()
+                else ".*";
+
+  const searchDir = MasonUpdate.getRegistryDir() + "/Bricks/";
+  const pattern   = compile(query, ignorecase=true);
+
+  var results : [1..0] string;
+  for dir in listdir(searchDir, files=false, dirs=true) {
+    if pattern.search(dir) {
+
+      const pkg = dir.replace("/", "");
+      const ver = findLatest(searchDir + dir);
+
+      results.push_back(pkg + " (" + ver.str() + ")");
+    }
+  }
+  for r in results.sorted() do writeln(r);
+
+  if results.size == 0 {
+    exit(1);
+  }
+}
+
+proc findLatest(packageDir) {
+  var ret = new VersionInfo(0, 0, 0);
+  const suffix = ".toml";
+
+  for fi in listdir(packageDir, files=true, dirs=false) {
+    assert(fi.endsWith(suffix));
+    const end = fi.length - suffix.length;
```

Comment:
Just a thought - [Path](http://chapel.cray.com/docs/master/modules/standard/Path.html) should really have procedures for extracting / dropping the file extension...